### PR TITLE
[SYCL] Add runtime library versioning

### PR DESCRIPTION
**Pull request by alexbatashev**
- On Linux runtime library now has version postfix
- On Windows version information is embedded into DLL
- Update ABI Policy Guide to reflect these changes

Signed-off-by: Alexander Batashev <alexander.batashev@intel.com>

### CODE REVIEW

**Comment by bso-intel** [comment]
Does this policy apply only to libsycl.so?
We also generate other runtime libraries such as libpi_opencl.so.

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,16 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 option(SYCL_ENABLE_WERROR "Treat all warnings as errors in SYCL project" OFF)
+option(SYCL_ADD_DEV_VERSION_POSTFIX "Adds -V postfix to version string" ON)
+
+set(SYCL_MAJOR_VERSION 0)
+set(SYCL_MINOR_VERSION 1)
+set(SYCL_PATCH_VERSION 0)
+set(SYCL_DEV_ABI_VERSION 0)
```
Comment (bso-intel) [inline sycl/CMakeLists.txt:14]: These numbers should be manually incremented every time we make some breaking or non-breaking changes?

Comment (alexbatashev) [reply sycl/CMakeLists.txt:14]: Only DEV_ABI version must be incremented, and only when breaking change happens. The rest of the numbers is changed once per release.

**Comment by alexbatashev** [comment]
> Does this policy apply only to libsycl.so?
> We also generate other runtime libraries such as libpi_opencl.so.

Only libsycl.so has versioning at the moment.

File: sycl/CMakeLists.txt (hunk 1 above)
Comment (bso-intel) [reply sycl/CMakeLists.txt:14]: Thanks for clarification.

**Pull request merged**